spring-batch-core/src/main/java/org/springframework/batch/core/StepContribution.java

```diff
@@ -191,6 +195,14 @@ public int getProcessSkipCount() {
 		return processSkipCount;
 	}
 
+	/**
+	 * Public getter for the parent step execution of this contribution.
+	 * @return parent step execution of this contribution
+	 */
+	public StepExecution getStepExecution() {
+		return stepExecution;
+	}
```

Comment:
we might need other things in addition to job/step name so I would stick to `StepExecution`.

Reply:
The more I think about this, the less I like it.  The `StepContribution` is basically supposed to represent the `StepExecution` so wrapping the `StepExecution` in the `StepContribution` just doesn't feel right to me.  I'd rather just copy the fields over into this class.  I get that it's more code, but I believe it's more clear on the intent.  If you agree with this, I can make the update on merge.

Reply:
> so wrapping the `StepExecution` in the `StepContribution` just doesn't feel right to me

We are not wrapping the `StepExecution` in the `StepContribution`, that would be wrong! We are just keeping a reference to the enclosing `StepExecution`. FTR, when we create a step execution from a job execution using [`JobExecution#createStepExecution`](https://github.com/spring-projects/spring-batch/blob/master/spring-batch-core/src/main/java/org/springframework/batch/core/JobExecution.java#L228), the returned `StepExecution` has a [reference to the enclosing job execution](https://github.com/spring-projects/spring-batch/blob/master/spring-batch-core/src/main/java/org/springframework/batch/core/StepExecution.java#L102) and a [getter for it](https://github.com/spring-projects/spring-batch/blob/master/spring-batch-core/src/main/java/org/springframework/batch/core/StepExecution.java#L331). Same thing  for the step contribution here, when we create a step contribution from a step executionusing [StepExecution#createStepContribution](https://github.com/spring-projects/spring-batch/blob/master/spring-batch-core/src/main/java/org/springframework/batch/core/StepExecution.java#L340-L342), we keep a reference to the enclosing step execution. This is probably better explained with a diagram:

![step-contribution 001](https://user-images.githubusercontent.com/1210553/55802895-73e06680-5ad9-11e9-9ae9-8bbb38a5fa35.png)

The dotted links were missing, I added them here. What I like about this approach is:

* Design consistency with Job Execution -> Step Execution
* Simplicity: I don't like passing job name, step name, etc everywhere in method signatures
* Proactive: we might need other infos in addition to job name and step name

If your concern is getting access to the step execution from the step contribution in `Tasklet#execute(StepContribution contribution, ChunkContext chunkContext)` and messing with it, then this would not be an issue since the step execution is accessible from the chunk context through `chunkContext.getStepContext().getStepExecution()`.